runtime-sdk/modules/contracts/src/abi/oasis/memory.rs

```diff
@@ -46,6 +46,19 @@ impl Region {
         }
     }
 
+    /// Dereferences a pointer to the region.
+    pub fn deref(memory: &wasm3::Memory<'_>, arg: u32) -> Self {
+        let arg = arg as usize;
+        let dst = memory.as_slice();
```

Comment:
This should validate that arg is within the memory region as otherwise the `dst` derefs below will panic. So this needs to return a `Result`.

Reply:
Done. 

Also I changed the existing `BadPointer` checks in this module from: `self.offset + self.length > memory.size()` to `self.offset + self.length >= memory.size()` as I think this is the correct check, as memory addresses are zero-based index?